packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -137,3 +163,55 @@ export function generateDataListEmptyState({
 
   return EmptyStateComponent;
 }
+
+export function sortSizeProp(sizeProp: Breakpoints[]) {
+  return sizeProp.sort(
+    (a, b) => BREAKPOINTS.indexOf(a) - BREAKPOINTS.indexOf(b),
+  );
+}
+
+export function useGridLayoutMediaQueries() {
+  const mediaQueries = BREAKPOINTS.reduce(
+    (previous, breakpoint) => ({
+      ...previous,
+      [breakpoint]: window.matchMedia(breakpointToMediaQuery(breakpoint)),
+    }),
+    {} as Record<Breakpoints, MediaQueryList>,
+  );
+
+  const initialMatches = BREAKPOINTS.reduce(
+    (previous, breakpoint) => ({
+      ...previous,
+      [breakpoint]: mediaQueries[breakpoint].matches,
+    }),
+    {} as Record<Breakpoints, boolean>,
+  );
+
+  const [matches, setMatches] = useState(initialMatches);
+  // Set up mediaQuery event handlers
+  useEffect(() => {
+    const handlers = BREAKPOINTS.reduce((previous, breakpoint) => {
+      const handler = (e: MediaQueryListEvent) =>
+        setMatches(previousMatches => ({
+          ...previousMatches,
+          [breakpoint]: e.matches,
+        }));
+      mediaQueries[breakpoint].addEventListener("change", handler);
```

Comment:
okay, this threw me off for a bit since it is more of a side effect than what reduce is meant to do.

should this be created outside of the effect and then foreach it to add the listeners like the return?

Reply:
```ts
const handlers = reduce // <--- here

useEffect(() => {
  const handlers = reduce // <--- or here
  
  handers.forEach(...) // add event listeners

  return () => {
    handers.forEach(...) // add event listeners
  }
})
```

My point here is to separate concerns and let that function inside the reduce do 1 thing. Right now it is doing 2 things and it is an easily missable thing down the line when a bug happens

Reply:
I see. That makes sense, I'll apply it